src/ir.cpp

```diff
@@ -18050,19 +18046,8 @@ static Error ir_make_type_info_value(IrAnalyze *ira, IrInstruction *source_instr
         case ZigTypeIdNull:
         case ZigTypeIdArgTuple:
         case ZigTypeIdOpaque:
-            *out = nullptr;
-            return ErrorNone;
-        default:
-            {
-                // Lookup an available value in our cache.
```

Comment:
I believe with a better fix as mentioned above, the cache will still function properly.

Reply:
Pretty sure this default case is never executed. It's a switch on `type_entry->id`, and all type entries are being listed as cases. I'm also pretty sure that the behavior of this code is wrong. Fall through to the `ZigTypeIdInt` case if we couldn't find anything in the cache seems wrong.

Reply:
I deleted this code, as it should never execute. We switch on `type_entry->id` and all `ZigTypeId`s have cases. It also seems wrong to fall through to the `ZigTypeIdInt` case if we didn't find anything in the cache. I could move the cache check out of the switch, which is probably was is actually intended.